tools/firtool/firtool.cpp

```diff
@@ -514,6 +520,12 @@ processBuffer(MLIRContext &context, TimingScope &ts, llvm::SourceMgr &sourceMgr,
     pm.nest<firrtl::CircuitOp>().nest<firrtl::FModuleOp>().addPass(
         firrtl::createInferReadWritePass());
 
+  if (lowerMemory)
```

Comment:
Cool, this runs before prefix and inline. So the instance and module names should just work. Nice.